design/external-names.md

```diff
@@ -0,0 +1,47 @@
+# External Names Service
+
+_Status_: Draft
+
+This document looks to allow Contour to process Kubernetes service types `ExternalName` so that traffic can be proxied to dns hosts rather than only pod endpoints.
+
+## Goals
+
+- Allow for Contour to proxy traffic via Envoy to external dns endpoints
+- Utilize built-in Kubernetes object types (i.e. ExternalName service)
+
+## Non-goals
+
+- `externalIPs` will not be used, users who want to proxy to custom IP's should build their own service/endpoint object
+- Utilize a new CRD or ConfigMap style mapping
+
+## Background
+
+Contour currently watches Kubernetes services and endpoints, streaming them to via Envoy xDS CDS & EDS endpoints.
+Requests to envoy are routed directly to Kubernetes pods (i.e. endpoints).
+This proposal looks to implement Kubernetes service type `ExternalName` which allows for an `externalName` which is represented as a valid RFC-1123 hostname.
```

Comment:
Possibly;

This proposal looks to add a second way to identify the members of a Envoy Cluster object, using a DNS entry to supply the cluster members rather than EDS.